perllib/FixMyStreet/App/Form/Waste/Garden/Modify.pm

```diff
@@ -65,12 +72,18 @@ has_page summary => (
         my $form = shift;
         my $c = $form->{c};
         my $data = $form->saved_data;
-        my $new_bins = $data->{bins_wanted} - $data->{current_bins};
+        my $current_bins = $data->{current_bins};
+        my $bin_count = $data->{bins_wanted};
+        my $new_bins = $bin_count - $current_bins;
         my $pro_rata = $c->cobrand->waste_get_pro_rata_cost( $new_bins, $c->stash->{garden_form_data}->{end_date});
-        my $total = $c->cobrand->garden_waste_cost($data->{bins_wanted});
+        my $cost_pa = $c->cobrand->garden_waste_cost_pa($bin_count);
+        my $cost_now_admin = $c->cobrand->garden_waste_new_bin_admin_fee($new_bins);
+        my $total = $cost_pa;
+        $pro_rata += $cost_now_admin;
```

Comment:
There's quite a lot of doing the bin cost assignment. I wonder if it might be worth extracting out into a method.

Reply:
Yeah, I wondered, but e.g. here it's the pro-rata cost which may (in a case where both pro-rata and admin fee are used) be different from annual.